hw/opentitan/ot_i2c.c

```diff
@@ -134,67 +137,107 @@ REG32(FDATA, 0x1cu)
 REG32(FIFO_CTRL, 0x20u)
     FIELD(FIFO_CTRL, RXRST, 0u, 1u)
     FIELD(FIFO_CTRL, FMTRST, 1u, 1u)
-    FIELD(FIFO_CTRL, RXILVL, 2u, 3u)
-    FIELD(FIFO_CTRL, FMTILVL, 5u, 2u)
     FIELD(FIFO_CTRL, ACQRST, 7u, 1u)
     FIELD(FIFO_CTRL, TXRST, 8u, 1u)
-REG32(FIFO_STATUS, 0x24u)
-    FIELD(FIFO_STATUS, FMTLVL, 0u, 7u)
-    FIELD(FIFO_STATUS, TXLVL, 8u, 7u)
-    FIELD(FIFO_STATUS, RXLVL, 16u, 7u)
-    FIELD(FIFO_STATUS, ACQLVL, 24u, 7u)
-REG32(OVRD, 0x28u)
+REG32(HOST_FIFO_CONFIG, 0x24u)
+    FIELD(HOST_FIFO_CONFIG, RX_THRESH, 0u, 12u)
+    FIELD(HOST_FIFO_CONFIG, FMT_THRESH, 16u, 12u)
+REG32(TARGET_FIFO_CONFIG, 0x28u)
+    FIELD(TARGET_FIFO_CONFIG, TX_THRESH, 0u, 12u)
+    FIELD(TARGET_FIFO_CONFIG, ACQ_THRESH, 16u, 12u)
+REG32(HOST_FIFO_STATUS, 0x2cu)
+    FIELD(HOST_FIFO_STATUS, FMTLVL, 0u, 12u)
+    FIELD(HOST_FIFO_STATUS, RXLVL, 16u, 12u)
+REG32(TARGET_FIFO_STATUS, 0x30u)
+    FIELD(TARGET_FIFO_STATUS, TXLVL, 0u, 12u)
+    FIELD(TARGET_FIFO_STATUS, ACQLVL, 16u, 12u)
+REG32(OVRD, 0x34u)
     FIELD(OVRD, TXOVRDEN, 0u, 1u)
     FIELD(OVRD, SCLVAL, 1u, 1u)
     FIELD(OVRD, SDAVAL, 2u, 1u)
-REG32(VAL, 0x2cu)
+REG32(VAL, 0x38u)
     FIELD(VAL, SCL_RX, 0u, 16u)
     FIELD(VAL, SDA_RX, 16u, 16u)
-REG32(TIMING0, 0x30u)
-    FIELD(TIMING0, THIGH, 0u, 16u)
-    FIELD(TIMING0, TLOW, 16u, 16u)
-REG32(TIMING1, 0x34u)
-    FIELD(TIMING1, T_R, 0u, 16u)
-    FIELD(TIMING1, T_F, 16u, 16u)
-REG32(TIMING2, 0x38u)
-    FIELD(TIMING2, TSU_STA, 0u, 16u)
-    FIELD(TIMING2, THD_STA, 16u, 16u)
-REG32(TIMING3, 0x3cu)
-    FIELD(TIMING3, TSU_DAT, 0u, 16u)
-    FIELD(TIMING3, THD_DAT, 16u, 16u)
-REG32(TIMING4, 0x40u)
-    FIELD(TIMING4, TSU_STO, 0u, 16u)
-    FIELD(TIMING4, T_BUF, 16u, 16u)
-REG32(TIMEOUT_CTRL, 0x44u)
-    FIELD(TIMEOUT_CTRL, VAL, 0u, 31u)
+REG32(TIMING0, 0x3cu)
+    FIELD(TIMING0, THIGH, 0u, 12u)
+    FIELD(TIMING0, TLOW, 16u, 12u)
+REG32(TIMING1, 0x40u)
+    FIELD(TIMING1, T_R, 0u, 12u)
+    FIELD(TIMING1, T_F, 16u, 12u)
+REG32(TIMING2, 0x44u)
+    FIELD(TIMING2, TSU_STA, 0u, 12u)
+    FIELD(TIMING2, THD_STA, 16u, 12u)
+REG32(TIMING3, 0x48u)
+    FIELD(TIMING3, TSU_DAT, 0u, 12u)
+    FIELD(TIMING3, THD_DAT, 16u, 12u)
+REG32(TIMING4, 0x4cu)
+    FIELD(TIMING4, TSU_STO, 0u, 12u)
+    FIELD(TIMING4, T_BUF, 16u, 12u)
```

Comment:
I think that these field sizes don't match the docs?

```suggestion
REG32(TIMING0, 0x3cu)
    FIELD(TIMING0, THIGH, 0u, 13u)
    FIELD(TIMING0, TLOW, 16u, 13u)
REG32(TIMING1, 0x40u)
    FIELD(TIMING1, T_R, 0u, 10u)
    FIELD(TIMING1, T_F, 16u, 9u)
REG32(TIMING2, 0x44u)
    FIELD(TIMING2, TSU_STA, 0u, 13u)
    FIELD(TIMING2, THD_STA, 16u, 13u)
REG32(TIMING3, 0x48u)
    FIELD(TIMING3, TSU_DAT, 0u, 9u)
    FIELD(TIMING3, THD_DAT, 16u, 13u)
REG32(TIMING4, 0x4cu)
    FIELD(TIMING4, TSU_STO, 0u, 13u)
    FIELD(TIMING4, T_BUF, 16u, 13u)
```